rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java

```diff
@@ -0,0 +1,392 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.Address;
+import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.bitcoinj.core.Context;
+import co.rsk.bitcoinj.core.ScriptException;
+import co.rsk.bitcoinj.core.TransactionInput;
+import co.rsk.bitcoinj.core.Utils;
+import co.rsk.bitcoinj.script.RedeemScriptParser;
+import co.rsk.bitcoinj.script.RedeemScriptParserFactory;
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.bitcoinj.script.ScriptBuilder;
+import co.rsk.bitcoinj.script.ScriptChunk;
+import co.rsk.bitcoinj.wallet.Wallet;
+import co.rsk.config.BridgeConstants;
+import co.rsk.peg.bitcoin.BitcoinUtils;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static org.ethereum.config.blockchain.upgrades.ConsensusRule.RSKIP293;
+
+/**
+ * @deprecated Methods included in this class are to be used only prior to the latest HF activation
+ * Use instead methods in {@link co.rsk.peg.PegUtils}
+ */
+@Deprecated
+public class PegUtilsLegacy {
+
+    private static final Logger logger = LoggerFactory.getLogger(PegUtilsLegacy.class);
+
+    private PegUtilsLegacy() {}
+
+    /**
+     * Legacy version for identifying if a tx is a pegout
+     * Use instead {@link co.rsk.peg.PegUtils#getTransactionType}
+     *
+     * @param tx
+     * @param federations
+     * @param activations
+     * @return true if it is a peg-out. Otherwise, returns false.
+     */
+    @Deprecated
+    public static boolean isPegOutTx(BtcTransaction tx, List<Federation> federations, ActivationConfig.ForBlock activations) {
```

Comment:
Check visibility of these functions? Are they used from anywhere else? (Check powpeg-node also). Maybe they can be private or protected, `getTransactionType` should probably take care of everything

Reply:
Yes, most of them can be changed to protected but not to private since many tests use them.  Only these two will remain public PegUtilsLegacy.isValidPegInTx and PegUtilsLegacy.isPegOutTx, because they are used from the powpeg-node.

Reply:
Well, we really shouldn't be testing private methods. But that's another story, please create a ticket to review this. Those methods should probably be private and be tested through getTransactionType